### DISCUSSION TITLE
Auto Adjust Screen Brightness

### DISCUSSION THREAD

**Discussion by ghost** [seerge/g-helper#1559] [Ideas]
Hey, I love this app and it's lightweight UI.

I think it would be great to add a feature to tweak the brightness when you are on battery or AC power as well. It seems Windows doesnt handle this natively anymore. At least on my 2023 G14 (4060).

I found this discussion here which had a powershell command to do so. I didn't use it entirely because I didnt want it to run constantly and readjust every 15s if I decided to turn mine up or down manually.

https://www.tenforums.com/graphic-cards/151916-guide-how-auto-adjust-brightness-when-running-battery-charging.html

Instead what I did, was use the command in a batch file and used windows task scheduler to run it every time I unlock the computer. That way, once I am on, I can adjust it as required. But if I forget that I took my laptop off AC and opened it in class, my brightness auto goes down to preserve battery. This is the command I put in a .bat and have task scheduler running on unlock to set to 10% / 60% for BAT / AC:

powershell -Command if( (Get-WmiObject -Class BatteryStatus -Namespace root\wmi -ComputerName "localhost").PowerOnLine ) { (Get-WmiObject -Namespace root/WMI -Class WmiMonitorBrightnessMethods).WmiSetBrightness(1,60) } else { (Get-WmiObject -Namespace root/WMI -Class WmiMonitorBrightnessMethods).WmiSetBrightness(1,10) }

Thanks!

**Comment by seerge** [comment]
@akni23 hello, you can set that inside windows power plan already 

https://superuser.com/questions/1250965/permanently-set-specific-brightness-percentage-for-power-plan :) 

If you are missing this setting (i.e. it's hidden) you need to unhide it first
https://www.tenforums.com/tutorials/108126-add-remove-display-brightness-power-options-windows.html

**Reply by ghost** [reply]
Hey @seerge thanks for the quick reply but I don’t think that works on Win 11. Not sure why they took that options out of Windows as it really impacts the battery life.

Note: I didn’t try to unhide it yet but that link does say it wont work after Win 10

**Reply by Kikimahe** [reply]
Hi, 

Yes unfortunately, Microsoft has removed this option. :-(

I tried to change several attribute values (2), in the different folders of
```
HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Power\PowerSettings\7516b95f-f776-4464-8c53-06167f40cc99\ 
Attributes DWORD / 1 = Remove 2 = Add
```

(No need to restart windows)
New options appear in display but not the one that interests us "display brightness" battery/plugged in.  ;-(
